bst_plugin/OM/process_nst_OM.m

```diff
@@ -393,394 +309,581 @@
 end
 
 
-function [montage_pairs,montage_weight] = compute_optimal_montage(head_vertices_coords,options)
+function [montage_pairs,montage_weight] = compute_optimal_montage(head_vertices_coords, options)
+    
+    %Define the cplex problem
+    [prob, options] = define_prob_simple(head_vertices_coords, options);
+   
+    cplex=Cplex(prob);
+    cplex.Model.sense = 'maximize';
+    cplex.Param.timelimit.Cur=300;
+
+    % Delete clone[number].log files created by Cplex
+    cplex.Param.output.clonelog.Cur = 0;
+
+    %Progress bar
+    bst_progress('start', 'Optimization','Running optimization with Cplex. May take several minutes (see matlab console) ...', 1, 1000);
+    
+    %======================================================================
+    % initial condition : find sources whose pairs have maximum energy then
+    % complete with detectors
+    % Uncomment the line to use initial condition
+    % cplex = init_solution(cplex, options.weight_tables, options.nH, options.nb_sources, options.nb_detectors);
+    %======================================================================
+  
+    results = cplex.solve;
+    if ~isfield(results, 'x')
+        bst_error(['OM computation failed  at Cplex step:', results.statusstring]);
+        return;
+    end
+   
+    addConstraint = 0;
+    if addConstraint
+       
+    end
+
+    bst_progress('stop');
 
-if ~isfield(options, 'nb_sources')
-    bst_error('Number of sources is required');
+    %Calculation of montage_pairs matrix and montage_weight vector
+    [montage_pairs, montage_weight] = montage_pairs_and_weight(results, options.nH, options.holder_distances, ...
```

Comment:
replace by montage_pairs_and_weight(results, options) and extract the different options in the function montage_pairs_and_weight